### ISSUE TITLE
Failure to overwrite settings with python lib usage

### ISSUE THREAD

**Issue by AlanCoding** [ansible/tower-cli#228]
I should be able to change the username, host, and password when executing a tower-cli command. I should still be able to do this when importing it in python and executing that same command.

That does not appear to be working. Consider the following, performing a list of users, giving a new value for `tower_host`.

```
>>> user_res.list(verbose=True, tower_host="http://localhost")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/alancoding/.virtualenvs/cli2/lib/python2.7/site-packages/ansible_tower_cli-3.0.1-py2.7.egg/tower_cli/models/base.py", line 732, in list
    response = self.read(**kwargs)
  File "/Users/alancoding/.virtualenvs/cli2/lib/python2.7/site-packages/ansible_tower_cli-3.0.1-py2.7.egg/tower_cli/models/base.py", line 516, in read
    r = client.get(url, params=kwargs)
  File "/Users/alancoding/.virtualenvs/cli2/lib/python2.7/site-packages/requests/sessions.py", line 487, in get
    return self.request('GET', url, **kwargs)
  File "/Users/alancoding/.virtualenvs/cli2/lib/python2.7/site-packages/ansible_tower_cli-3.0.1-py2.7.egg/tower_cli/api.py", line 137, in request
    'Right now it is: "%s".' % settings.host
tower_cli.utils.exceptions.ConnectionError: There was a network error of some kind trying to connect to Tower.

The most common  reason for this is a settings issue; is your "host" value in `tower-cli config` correct?
Right now it is: "http://0.0.0.0:8013".
```

It uses the _old_ value of the host. This is a component of the issue #149, but here, it is using the outright incorrect value.

We will probably be able to unit test this bug.

**Comment by AlanCoding** [comment]
This is still one of my top priorities, but it's too much to cram into the 3.1.0 release.

**Comment by AlanCoding** [comment]
believed to be solved by #266

**Comment by jangsutsr** [comment]
Issue solved. Patch will ship with 3.1.1 release.
